wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+Forward.swift

```diff
@@ -167,6 +167,17 @@ extension ConversationContentViewController {
             shareViewController.showPreview = traitCollection.horizontalSizeClass != .regular
         }
     }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+
+        coordinator.animate(alongsideTransition: nil) { _ in
+            self.dataSource.resetSectionControllers()
+            self.dataSource.reloadSections(newSections: self.dataSource.calculateSections())
+            self.tableView.reloadData()
+        }
+    }
```

Comment:
question: when does this happen? and what are we doing exactly?

Reply:
This is to update iPad app when you hide or show sidebar. Content area is different so we need to redraw, anyway it looks not ideal on iPad but at least it updates. I tried to improve but it was not updating easily and I had not time to fix it properly

Reply:
todo: leave a comment to explain this in code, if you think there could be a better solution you can also create a follow up ticket.

```suggestion
        super.viewWillTransition(to: size, with: coordinator)
        // TODO: [WPB-XXX] this is to update the cells on iPad when sidebar 
        // is hidden or shown. This is a quick fix for now. To improve later
        coordinator.animate(alongsideTransition: nil) { _ in
            self.dataSource.resetSectionControllers()
            self.dataSource.reloadSections(newSections: self.dataSource.calculateSections())
            self.tableView.reloadData()
        }
    }
```

Reply:
Improved